SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -318,14 +319,16 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject&
 					}
 				}
 
+				TOptional<Trace_SpanId> SpanId;
 				if (bEventTracerEnabled)
 				{
 					Worker_EntityId EntityId = Channel.GetEntityId();
 
 					Trace_SpanId CauseSpanId = EventTracer->GetSpanId(EntityComponentId(EntityId, ComponentId));
-					TOptional<Trace_SpanId> SpanId = EventTracer->CreateSpan(&CauseSpanId, 1);
+					SpanId = EventTracer->CreateSpan(&CauseSpanId, 1);
```

Comment:
Are we guaranteed to always have a SpanId for the pair EntityId/ComponentId here? What about if probabilistic sampling was enabled?

Reply:
So we should always have some SpanId in the cache for a entity component at this point. It might be possible to be an invalid SpanId due to sampling. If that is the case we should be fine. Perhaps we should check for invalid SpanIds and just not bother tracing the event. "Or this pair not being traced" how could this happen?

Reply:
We could avoid storing those spans if they are not set and use existence to determine whether a cause is valid here - just saves maintaining a large map of entity/components. I guess once sampling comes in this map would be a lot less full so can save some memory and complexity updating it (unless I missed something!)

Reply:
Nice I will add this.